backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +71,97 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author_with_config(author_data, matcher_config):
+    matched_records = (
+        matched_record["inner_hits"]["authors"]["hits"]["hits"]
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    )
+    matched_authors_refs = set()
+    for record_authors in matched_records:
+        matched_authors_refs.update(
+            author["_source"]["record"]["$ref"]
+            for author in record_authors
+            if "record" in author["_source"]
+        )
+    return matched_authors_refs
+
+
+def assign_reference_to_author(matched_authors_refs, author):
+    if matched_authors_refs and len(matched_authors_refs) == 1:
```

Comment:
are you ever passing `None`? otherwise this works
```suggestion
    if len(matched_authors_refs) == 1:
```